packages/gitbook/src/components/DocumentView/Blocks.tsx

```diff
@@ -49,22 +50,35 @@ type UnwrappedBlocksProps<TBlock extends DocumentBlock> = DocumentContextProps &
 export function UnwrappedBlocks<TBlock extends DocumentBlock>(props: UnwrappedBlocksProps<TBlock>) {
     const { nodes, blockStyle, ...contextProps } = props;
 
+    let isOffscreen = false;
+
     return (
         <>
-            {nodes.map((node) => (
-                <Block
-                    key={node.key}
-                    block={node}
-                    style={[
-                        'w-full mx-auto decoration-primary/6',
-                        node.data && 'fullWidth' in node.data && node.data.fullWidth
-                            ? 'max-w-screen-xl'
-                            : 'max-w-3xl',
-                        blockStyle,
-                    ]}
-                    {...contextProps}
-                />
-            ))}
+            {nodes.map((node) => {
+                isOffscreen =
+                    isOffscreen ||
+                    isBlockOffscreen({
+                        document: props.document,
+                        block: node,
+                        ancestorBlocks: props.ancestorBlocks,
+                    });
```

Comment:
I get that you're avoiding to recompute if true but are all blocks unwrapped at the same time? can it not change?

Reply:
The blocks are in order, so if a block is offscreen, the next one will be as well.